routing/helicopter_router.cpp

```diff
@@ -0,0 +1,139 @@
+#include "routing/helicopter_router.hpp"
+#include "routing/route.hpp"
+#include "routing/routing_helpers.hpp"
+
+namespace routing
+{
+using namespace std;
+
+void HelicopterRouter::ClearState()
+{
+}
+
+void HelicopterRouter::SetGuides(GuidesTracks && guides) { /*m_guides = GuidesConnections(guides);*/ }
+
+
+/* Helicopter router doesn't read roads graph and uses only checkpoints to build a route.
+
+      1-----------2
+     /             \
+    /               \
+   /                 F
+  S
+
+  For example we need to build route from start (S) to finish (F) with intermidiate
+  points (1) and (2).
+  Target route should have parameters:
+
+  m_geometry      = [S, S, 1, 1, 2, 2, F, F]
+  m_routeSegments = [S, 1, 1, 2, 2, F, F]
+  m_subroutes     =
+    +-------+--------+-----------------+---------------+
+    | start | finish | beginSegmentIdx | endSegmentIdx |
+    +-------+--------+-----------------+---------------+
+    | S     | 1      | 0               | 1             |
+    | 1     | 1      | 1               | 2             |
+    | 1     | 2      | 2               | 3             |
+    | 2     | 2      | 3               | 4             |
+    | 2     | F      | 4               | 6             |
+    | 2     | F      | 4               | 6             |
+    +-------+--------+-----------------+---------------+
+
+  Constraints:
+  *  m_geometry.size() == 2 * checkpoints.size()
+  *  m_routeSegments.size() == m_geometry.size()-1
+  *  m_subroutes.size() == m_routeSegments.size()-1
+
+ */
+RouterResultCode HelicopterRouter::CalculateRoute(Checkpoints const & checkpoints,
+                                                      m2::PointD const & startDirection,
+                                                      bool adjustToPrevRoute,
+                                                      RouterDelegate const & delegate, Route & route)
+{
+  vector<m2::PointD> const & points = checkpoints.GetPoints();
+  geometry::Altitude const mockAltitude = 0;
+  vector<RouteSegment> routeSegments;
+  vector<double> times;
```

Comment:
size_t const count = points.size();
ASSERT(count > 0, ());

And use it below.

Reply:
Introduced variable.